File: dotnet-server/Hubs/HubEvents.cs
namespace Dotnet.Server.Hubs;

public static class HubEvents
{
    //GameSettings
    public const string LoadGameSettings = "LoadGameSettings";
    public const string SetAbstractNouns = "SetAbstractNouns";
    public const string SetDrawingTimeSeconds = "SetDrawingTimeSeconds";
    public const string SetRoundsCount = "SetRoundsCount";
    public const string SetWordLanguage = "SetWordLanguage";
    public const string OnLoadGameSettings = "OnLoadGameSettings";
    public const string OnSetAbstractNouns = "OnSetAbstractNouns";
    public const string OnSetDrawingTimeSeconds = "OnSetDrawingTimeSeconds";
    public const string OnSetRoundsCount = "OnSetRoundsCount";
    public const string OnSetWordLanguage = "OnSetWordLanguage";

    //Chat
    public const string SendChatMessage = "SendChatMessage";
    public const string LoadChatMessages = "LoadChatMessages";
    public const string OnSendChatMessage = "OnSendChatMessage";
    public const string OnLoadChatMessages = "OnLoadChatMessages";
    public const string OnSendAnnouncement = "OnSendAnnouncement";

    //Game
    public const string JoinGame = "JoinGame";
    public const string LeaveGame = "LeaveGame";
    public const string FindPlayerHost = "FindPlayerHost";
    public const string OnJoinGame = "OnJoinGame";
    public const string OnJoinGameError = "OnJoinGameError";
    public const string OnGameProblem = "OnGameProblem";
    public const string OnFindPlayerHost = "OnFindPlayerHost";
    public const string OnEndGame = "OnEndGame";

    //GameState
    public const string StartGame = "StartGame";
    public const string GetSecretWord = "GetSecretWord";
    public const string OnStartGame = "OnStartGame";
    public const string OnGetSecretWord = "OnGetSecretWord";
    public const string OnRequestSecretWord = "OnRequestSecretWord";
    public const string OnUpdatePlayerScores = "OnUpdatePlayerScores";
    public const string OnUpdateTimer = "OnUpdateTimer";
    public const string OnUpdateTimerVisibility = "OnUpdateTimerVisibility";
    public const string OnGameFinished = "OnGameFinished";
    public const string OnUpdateDrawingPlayer = "OnUpdateDrawingPlayer";
    public const string OnUpdateCurrentRound = "OnUpdateCurrentRound";
    
    // Canvas
    public const string LoadCanvas = "LoadCanvas";
    public const string DrawOnCanvas = "DrawOnCanvas";
    public const string ClearCanvas = "ClearCanvas";
    public const string OnLoadCanvas = "OnLoadCanvas";
    public const string OnDrawOnCanvas = "OnDrawOnCanvas";
    public const string OnClearCanvas = "OnClearCanvas";
    public const string OnSetCanvasText = "OnSetCanvasText";
}

File: react-app/src/hub/HubEvents.ts
class HubEvents {
  //GameSettings
  static LoadGameSettings: string = "LoadGameSettings";
  static setAbstractNouns: string = "SetAbstractNouns";
  static setDrawingTimeSeconds: string = "SetDrawingTimeSeconds";
  static setRoundsCount: string = "SetRoundsCount";
  static setWordLanguage: string = "SetWordLanguage";
  static onLoadGameSettings: string = "OnLoadGameSettings";
  static onSetAbstractNouns: string = "OnSetAbstractNouns";
  static onSetDrawingTimeSeconds: string = "OnSetDrawingTimeSeconds";
  static onSetRoundsCount: string = "OnSetRoundsCount";
  static onSetWordLanguage: string = "OnSetWordLanguage";

  //Chat
  static sendChatMessage: string = "SendChatMessage";
  static loadChatMessages: string = "LoadChatMessages";
  static onSendChatMessage: string = "OnSendChatMessage";
  static onLoadChatMessages: string = "OnLoadChatMessages";
  static onSendAnnouncement: string = "OnSendAnnouncement";

  //Game
  static joinGame: string = "JoinGame";
  static leaveGame: string = "LeaveGame";
  static findPlayerHost: string = "FindPlayerHost";
  static onGameProblem: string = "OnGameProblem";
  static onJoinGame: string = "OnJoinGame";
  static onJoinGameError: string = "OnJoinGameError";
  static onFindPlayerHost: string = "OnFindPlayerHost";
  static onEndGame: string = "OnEndGame";

  //GameState
  static startGame: string = "StartGame";
  static getSecretWord: string = "GetSecretWord";
  static onStartGame: string = "OnStartGame";
  static onGetSecretWord: string = "OnGetSecretWord";
  static onRequestSecretWord: string = "OnRequestSecretWord";
  static onUpdatePlayerScores: string = "OnUpdatePlayerScores";
  static onUpdateTimer: string = "OnUpdateTimer";
  static onUpdateTimerVisibility: string = "OnUpdateTimerVisibility";
  static onUpdateDrawingPlayer: string = "OnUpdateDrawingPlayer";
  static onUpdateCurrentRound: string = "OnUpdateCurrentRound";
  static onUpdateCorrectGuessPlayerUsernames: string = "OnUpdateCorrectGuessPlayerUsernames";

  //Canvas
  static loadCanvas: string = "LoadCanvas";
  static drawOnCanvas: string = "DrawOnCanvas";
  static clearCanvas: string = "ClearCanvas";
  static undoLine: string = "UndoLine";
  static onLoadCanvas: string = "OnLoadCanvas";
  static onDrawOnCanvas: string = "OnDrawOnCanvas";
  static onClearCanvas: string = "OnClearCanvas";
  static OnSetCanvasText: string = "OnSetCanvasText";

  //Account
  static createSession = "CreateSession";
  static endSession = "EndSession";  
  static onSessionEnded = "OnSessionEnded";
  static onUpdateAccountScore: string = "OnUpdateAccountScore";

}

export default HubEvents;